src/Transport/Config/AzureServiceBusTransportInfrastructure.cs

```diff
@@ -27,7 +27,19 @@ public override EndpointInstance BindToLocalEndpoint(EndpointInstance instance)
 
         public override string ToTransportAddress(LogicalAddress logicalAddress)
         {
-            return logicalAddress.ToString();
+            var queue = new StringBuilder(logicalAddress.QueueName);
```

Comment:
Because the actual code depends on the transport. I mean, different transport can use different separators for qualifiers and discriminators (e.g. SQL) and can use additional properties (MSMQ, SQL)